libs/wire-subsystems/src/Wire/DomainRegistrationStore/Cassandra.hs

```diff
@@ -24,10 +25,35 @@ interpretDomainRegistrationStoreToCassandra casClient =
     embed @IO . runClient casClient . \case
       UpsertInternal dr -> upsertImpl dr
       LookupInternal domain -> lookupImpl domain
+      LookupByTeamInternal tid -> lookupByTeamInternalImpl tid
       DeleteInternal domain -> deleteImpl domain
 
+lookupByTeamInternalImpl :: (MonadClient m) => TeamId -> m [StoredDomainRegistration]
+lookupByTeamInternalImpl tid = do
+  domains <- lookupTeamDomains tid
+  fmap asRecord <$> retry x1 (query cql (params LocalQuorum (Identity domains)))
+  where
+    cql :: PrepQuery R (Identity [DomainKey]) (TupleType StoredDomainRegistration)
+    cql = "SELECT domain, domain_redirect, team_invite, idp_id, backend_url, team, dns_verification_token, ownership_token_hash, authorized_team FROM domain_registration WHERE domain IN ?"
```

Comment:
These `IN` queries are bad for performance. In cases like this, we normally run separate queries for each domain in parallel.